x/auth/ante/unordered.go

```diff
@@ -62,7 +78,39 @@ func (d *UnorderedTxDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, _ bool, ne
 		return ctx, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "unordered tx ttl exceeds %d", d.maxUnOrderedTTL)
 	}
 
-	txHash := sha256.Sum256(ctx.TxBytes())
+	// in order to create a deterministic hash based on the tx, we need to hash the contents of the tx with signature
+	// Get a Buffer from the pool
+	buf := bufPool.Get().(*bytes.Buffer)
```

Comment:
wondering if using this primitive will/can create issues for proving 🤔

Reply:
mh, i don't think this would work. if block execution fails a tx because it's not part of tx manager then this in theory means that tx manager state needs to be proven (merklized).